test/controllers/admin/calendars_controller_test.rb

```diff
@@ -4,17 +4,21 @@
 
 class Admin::CalendarControllerTest < ActionDispatch::IntegrationTest
   setup do
-    @root = create(:root)
-    @neighbourhood_admin = create(:neighbourhood_admin)
-    @partner_admin = create(:partner_admin)
+    VCR.use_cassette(:import_test_calendar) do
```

Comment:
The problem is that in the calendar model there is [this](https://github.com/geeksforsocialchange/PlaceCal/blob/428a0a7351347aaf8e81b303752a9a40b25c3f73/app/models/calendar.rb#L98) code that is triggered before save (when source has changed value). This then instantiates a CalendarImporter object which causes it to run [this](https://github.com/geeksforsocialchange/PlaceCal/blob/428a0a7351347aaf8e81b303752a9a40b25c3f73/app/jobs/calendar_importer/calendar_importer.rb#L24) validation. But if you look down the code to [here](https://github.com/geeksforsocialchange/PlaceCal/blob/428a0a7351347aaf8e81b303752a9a40b25c3f73/app/jobs/calendar_importer/calendar_importer.rb#L51) it swallows all the exceptions. which is bad.

in order to fulfil the acceptance criteria i had to refactor this bit of code so it only happens in one place and it only tests for failure conditions that it is concerned with.

the pain was that when taking out this `rescue StandardError` our test suite went on fire. not because i had broken something but because now we weren't blanket suppressing exceptions any more all the bugs in our test suite suddenly lit up.

Reply:
(this also led to a massive problem with our test factories)